File: DCP-1841/solution.kt
data class WebsiteSimilarity(
    val website1: String,
    val website2: String,
    val similarity: Double
)

fun findMostSimilarWebsites(visits: List<Pair<String, Int>>, k: Int): List<Pair<String, String>> {
    // Create a map of website to set of users
    val websiteUsers = visits.groupBy(
        keySelector = { it.first },
        valueTransform = { it.second }
    ).mapValues { it.value.toSet() }
    
    // Calculate similarities between all pairs
    val similarities = mutableListOf<WebsiteSimilarity>()
    val websites = websiteUsers.keys.toList()
    
    for (i in websites.indices) {
        for (j in i + 1 until websites.size) {
            val site1 = websites[i]
            val site2 = websites[j]
            
            val users1 = websiteUsers[site1]!!
            val users2 = websiteUsers[site2]!!
            
            // Calculate Jaccard similarity
            val intersection = users1.intersect(users2).size
            val union = users1.union(users2).size
            val similarity = intersection.toDouble() / union.toDouble()
            
            similarities.add(WebsiteSimilarity(site1, site2, similarity))
        }
    }
    
    // Sort by similarity and take top k
    return similarities
        .sortedByDescending { it.similarity }
        .take(k)
        .map { Pair(it.website1, it.website2) }
}

File: DCP-1841/solution.swift
// Problem 1841 (k pairs with greatest similarity)

import Foundation

struct WebsiteSimilarity {
    let website1: String
    let website2: String
    let similarity: Double
}

func findMostSimilarWebsites(_ visits: [(String, Int)], k: Int) -> [(String, String)] {
    // Create a dictionary of website to set of users
    var websiteUsers: [String: Set<Int>] = [:]
    
    // Group users by website
    for (website, user) in visits {
        websiteUsers[website, default: Set()].insert(user)
    }
    
    // Calculate similarities between all pairs
    var similarities: [WebsiteSimilarity] = []
    let websites = Array(websiteUsers.keys)
    
    for i in 0..<websites.count {
        for j in (i + 1)..<websites.count {
            let site1 = websites[i]
            let site2 = websites[j]
            
            let users1 = websiteUsers[site1]!
            let users2 = websiteUsers[site2]!
            
            // Calculate Jaccard similarity
            let intersection = users1.intersection(users2).count
            let union = users1.union(users2).count
            let similarity = Double(intersection) / Double(union)
            
            similarities.append(WebsiteSimilarity(
                website1: site1,
                website2: site2,
                similarity: similarity
            ))
        }
    }
    
    // Sort by similarity and take top k
    return similarities
        .sorted { $0.similarity > $1.similarity }
        .prefix(k)
        .map { ($0.website1, $0.website2) }
}